vpn/muxer.go

```diff
@@ -369,8 +414,9 @@ func (m *muxer) readAndLoadRemoteKey() error {
 		return fmt.Errorf("%w: %s", ErrBadHandshake, err)
 	}
 
-	// Parse and store the useful parts of the remote options.
-	m.tunnel = parseRemoteOptions(m.tunnel, opts)
+	// Parse and update the useful fields from the remote options (mtu).
+	ti := newTunnelInfoFromRemoteOptionsString(remoteOptStr)
```

Comment:
could parsing fail? do we care about that?

Reply:
I'm tempted to say I expect a well-behaved server. If we fail to parse correctly, I return a null tunnelInfo object.

Reply:
examining it more closely, I found a index-out-of-range panic, fixed and added a test case.